src/js/trove/multiple-regression.js

```diff
@@ -0,0 +1,182 @@
+({
+  requires: [],
+  nativeRequires: ["pyret-base/js/namespace"],
+  provides: {
+    values: {
+      "multiple-regression": "tany"
+    }
+  },
+  theModule: function(runtime, namespace, uri) {
+
+    function makeColumnMatrix(y_s) {
+      let n = y_s.length;
+      let M = new Array(n);
+      for (let i = 0; i < n; i++) {
+        M[i][0] = y_s[i];
+      }
+      return M;
+    }
+
+    function matrixTranspose(m) {
+      let numrows = m.length;
+      let numcols = m[0].length;
+      let mT = new Array(numcols);
+      for (let i = 0; i < numcols; i++) {
+        mT[i] = new Array(m);
+        for (let j = 0; j < numrows; j++) {
+          mT[i][j] = m[j][i];
+        }
+      }
+      return mT;
+    }
+
+    function matrixDeterminant(m) {
+      let size = m.length;
+      if (size === 0) {
+        return 0;
+      }
+      if (m[0].length !== size) {
+        return 0;
+      }
+      if (size === 1) {
+        return m[0][0];
+      }
+      if (size === 2) {
```

Comment:
Here, and in _all_ the arithmetic below: doesn't this need to use the jsnums library, rather than assume you've got JS doubles?  Or, if you're _relying_ on `num-to-fixnum` below, is that a deliberate choice in the stats library to switch entirely to roughnums?  (If a user supplied a list of exact fractions, say, shouldn't the estimator also be exactly rational?)

Reply:
Given real-life data, it is unlikely that the coefficients of the predictor function will be anything but roughnums, no? And in the very rare cases they end up being exact rationals, "mistaking" them for roughnums is not catastrophic.